core/src/main/scala/org/apache/spark/SSLOptions.scala

```diff
@@ -0,0 +1,115 @@
+      keyPassword.foreach(sslContextFactory.setKeyManagerPassword)
+      protocol.foreach(sslContextFactory.setProtocol)
+      sslContextFactory.setIncludeCipherSuites(enabledAlgorithms.toSeq: _*)
+
+      Some(sslContextFactory)
+    } else {
+      None
+    }
+  }
+
+  /**
+   * Creates an Akka configuration object which contains all the SSL settings represented by this
+   * object. It can be used then to compose the ultimate Akka configuration.
+   */
+  def createAkkaConfig: Option[Config] = {
+    import scala.collection.JavaConversions._
+    if (enabled) {
+      Some(ConfigFactory.empty()
+        .withValue("akka.remote.netty.tcp.security.key-store",
+          ConfigValueFactory.fromAnyRef(keyStore.map(_.getAbsolutePath).getOrElse("")))
+        .withValue("akka.remote.netty.tcp.security.key-store-password",
+          ConfigValueFactory.fromAnyRef(keyStorePassword.getOrElse("")))
+        .withValue("akka.remote.netty.tcp.security.trust-store",
+          ConfigValueFactory.fromAnyRef(trustStore.map(_.getAbsolutePath).getOrElse("")))
+        .withValue("akka.remote.netty.tcp.security.trust-store-password",
+          ConfigValueFactory.fromAnyRef(trustStorePassword.getOrElse("")))
+        .withValue("akka.remote.netty.tcp.security.key-password",
+          ConfigValueFactory.fromAnyRef(keyPassword.getOrElse("")))
+        .withValue("akka.remote.netty.tcp.security.random-number-generator",
+          ConfigValueFactory.fromAnyRef(""))
+        .withValue("akka.remote.netty.tcp.security.protocol",
+          ConfigValueFactory.fromAnyRef(protocol.getOrElse("")))
+        .withValue("akka.remote.netty.tcp.security.enabled-algorithms",
+          ConfigValueFactory.fromIterable(enabledAlgorithms.toSeq))
+        .withValue("akka.remote.netty.tcp.enable-ssl",
+          ConfigValueFactory.fromAnyRef(true)))
+    } else {
+      None
+    }
+  }
+
+}
+
+object SSLOptions extends Logging {
+
+  /**
+   * Resolves SSLOptions settings from a given Spark configuration object at a given namespace.
+   * The parent directory of that location is used as a base directory to resolve relative paths
+   * to keystore and truststore.
+   */
+  def parse(conf: SparkConf, ns: String): SSLOptions = {
+    val enabled = conf.getBoolean(s"spark.ssl.enabled", defaultValue = false)
+    val keyStore = Try(conf.get(s"$ns.keyStore")).toOption.map(new File(_))
+    val keyStorePassword = Try(conf.get(s"$ns.keyStorePassword")).toOption
+    val keyPassword = Try(conf.get(s"$ns.keyPassword")).toOption
+    val trustStore = Try(conf.get(s"$ns.trustStore")).toOption.map(new File(_))
+    val trustStorePassword = Try(conf.get(s"$ns.trustStorePassword")).toOption
+    val protocol = Try(conf.get(s"$ns.protocol")).toOption
+    val enabledAlgorithms = Try(conf.get(s"$ns.enabledAlgorithms")).toOption
+      .map(_.trim.dropWhile(_ == '[')
+      .takeWhile(_ != ']')).map(_.split(",").map(_.trim).toSet)
```

Comment:
No, it will not, because the whole string is trimmed.

I wanted it to be compatible with Akka way of passing lists of values. But, now I see that it doesn't make much sense.